packages/medusa/src/services/cart.ts

```diff
@@ -1755,7 +1755,7 @@ class CartService extends TransactionBaseService {
     return await this.atomicPhase_(
       async (transactionManager: EntityManager) => {
         const cart = await this.retrieve(cartId, {
```

Comment:
thought(non-blocking): `retrieveWithTotals` gives better performance; might be worth switching it out; might be part of @adrien2p's work though